src/modules/user/user.service.ts

```diff
@@ -1,28 +1,103 @@
-import { Injectable } from '@nestjs/common';
+import { UpdateProfileDto } from './dto/requests/update-profile.dto';
+import {
+  BadRequestException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
 import { User } from './entities/user.entity';
+import { Repository } from 'typeorm';
 
 @Injectable()
 export class UserService {
-  constructor(@InjectRepository(User) private userRepo: User) {}
+  constructor(
+    @InjectRepository(User) private userRepository: Repository<User>,
+  ) {}
 
-  create() {
-    return 'This action adds a new user';
-  }
+  async getAllUsers() {
+    const users = await this.userRepository.find({
+      select: [
+        'id',
+        'email',
+        'firstName',
+        'lastName',
+        'role',
+        'createdAt',
+        'updatedAt',
+      ],
+    });
+    if (!users) {
+      throw new NotFoundException('Users are empty');
```

Comment:
I think you don't have to throw an error when no users are found. You can just return empty array

Reply:
I thought that it makes working in FE more comfortable.
For example, if you haven't users you will show text from the error message.

But for now i resolve it as you said.